File: javascript/fibonacci.js
function fibonacci(num) {
  if (num < 2) {
    return num;
  }

  let previousVals = [0, 1]

  for (let i = 0; i < num-1; ++i) {
    const newSum = previousVals[0] + previousVals[1];
    previousVals = [previousVals[1], newSum];
  }

  return previousVals[1];
}

//takes in a number as an argument
//for each number, add together all previous numbers (which is n value)
//returns n value


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
 previous_vals = [0, 1]

 return previous_vals[num] if num < 2

 (num-1).times do
  new_sum = previous_vals[0] + previous_vals[1]
  previous_vals = [previous_vals[1], new_sum]
 end

 previous_vals.last

end

# takes in a number as an argument
# for each number, add together all previous numbers 
# returns n value based on previous numbers



if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
